### PR TITLE
[CI] Separate pipeline for packageing elastic-agent v 7.17

### PR DESCRIPTION
**Pull request by pazone**
## Proposed commit message

Separate packaging pipeline for elastic-agent v7.17

Relates: https://github.com/elastic/ingest-dev/issues/2644

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -82,7 +82,7 @@ spec:
     kind: Pipeline
     metadata:
       name: beats-metricbeat
-      description: 'Beats Metricbeat pipeline'
+      description: "Beats Metricbeat pipeline"
     spec:
       branch_configuration: "main 7.* 8.* v7.* v8.*"
```
Comment (v1v) [inline catalog-info.yaml:87]: ditto

File: catalog-info.yaml [hunk 2]
```diff
@@ -310,9 +310,9 @@ spec:
       name: beats-libbeat
       description: "Beats libbeat pipeline"
     spec:
-#      branch_configuration: "main 7.17 8.* v7.17 v8.*" TODO: temporarily commented to build PRs from forks
+      #      branch_configuration: "main 7.17 8.* v7.17 v8.*" TODO: temporarily commented to build PRs from forks
```
Comment (v1v) [inline catalog-info.yaml:313]: ditto

File: catalog-info.yaml [hunk 3]
```diff
@@ -355,9 +355,9 @@ spec:
       name: beats-packetbeat
       description: "Beats packetbeat pipeline"
     spec:
-#      branch_configuration: "main 7.17 8.* v7.17 v8.*" TODO: temporarily commented to build PRs from forks
+      #      branch_configuration: "main 7.17 8.* v7.17 v8.*" TODO: temporarily commented to build PRs from forks
```
Comment (v1v) [inline catalog-info.yaml:358]: ditto

File: catalog-info.yaml [hunk 4]
```diff
@@ -413,9 +413,53 @@ spec:
...
+        publish_commit_status: false # do not update status of commits for this pipeline
+        build_tags: false
+        build_branches: false
+        filter_enabled: true
+        filter_condition: >-
+          build.pull_request.id == null || (build.creator.name == 'elasticmachine' && build.pull_request.id != null)
+      cancel_intermediate_builds: true
+      cancel_intermediate_builds_branch_filter: "!main"
+      skip_intermediate_builds: true
+      skip_intermediate_builds_branch_filter: "!main"
```
Comment (v1v) [inline catalog-info.yaml:458]: If `7.17` then cannot be cancelled?
```suggestion
      cancel_intermediate_builds_branch_filter: "!7.17"
      skip_intermediate_builds: true
      skip_intermediate_builds_branch_filter: "!7.17"
```

File: .buildkite/x-pack/elastic-agent/pipeline.xpack.elastic-agent.package.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+
+# This pipeline is only for 7.17 branch. See catalog-info.yml
+steps:
+  - label: "Example test"
+    command: echo "Hello!"
```
Comment (belimawr) [inline .buildkite/x-pack/elastic-agent/pipeline.xpack.elastic-agent.package.yml:6]: Was it forgotten here? Or is it here as a placeholder that `catalog-info.yml` can reference?

**Review by belimawr — COMMENTED** [review]
I know little about the CI part of Beats, but LGTM. Anyway I'll wait for the points raised by @v1v to be addressed and CI to be green.

File: catalog-info.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (pazone) [reply catalog-info.yaml:458]: I think yes. The pipeline is only for `7.17`

File: .buildkite/x-pack/elastic-agent/pipeline.xpack.elastic-agent.package.yml (hunk 5 above)
Comment (pazone) [reply .buildkite/x-pack/elastic-agent/pipeline.xpack.elastic-agent.package.yml:6]: That's right

**Comment by pazone** [comment]
@v1v Corrected the branch configuration

**Review by v1v — CHANGES_REQUESTED** [review]
There are merge conflicts

File: catalog-info.yaml [hunk 6]
```diff
@@ -97,11 +97,18 @@ spec:
...
       cancel_intermediate_builds: true
-      cancel_intermediate_builds_branch_filter: '!main !7.* !8.*'
+      cancel_intermediate_builds_branch_filter: "!main !7.* !8.*"
       skip_intermediate_builds: true
+<<<<<<< HEAD
+      skip_intermediate_builds_branch_filter: "!main !7.* !8.*"
+      # TODO uncomment this environment variable when pipeline definition is updated
+      # env:
+      #   ELASTIC_PR_COMMENTS_ENABLED: 'true'
+=======
```
[the remark this answers is not in the document]
Comment (pazone) [reply catalog-info.yaml]: oh. Fixed.

File: catalog-info.yaml [hunk 7]
```diff
@@ -658,3 +658,47 @@ spec:
...
+      url: https://buildkite.com/elastic/beats-xpack-elastic-agent-package-7-17
+spec:
+  type: buildkite-pipeline
+  owner: group:ingest-fp
+  system: buildkite
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: elastic-agent-package-717
```
Comment (v1v) [inline catalog-info.yaml:680]: to match the same pattern in `metadata.name`
```suggestion
      name: elastic-agent-package-7-17
```

**Review by v1v — COMMENTED** [review]
I'm a bit confused about the title and the changes.

If this is only for `7.17`, what's the reason `main` and `8.x` branches are in the list of supported branches?

**Comment by pazone** [comment]
@v1v the pipeline will be triggered remotely. Removed other branches from configuration just in case

File: catalog-info.yaml [hunk 8]
```diff
@@ -648,13 +648,57 @@ spec:
...
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: elastic-agent-package-7-17
+      description: Buildkite pipeline for packaging Elastic Agent package
+    spec:
+      repository: elastic/elastic-agent
+      pipeline_file: ".buildkite/x-pack/elastic-agent/pipeline.xpack.elastic-agent.package.yml"
+      provider_settings:
```
Comment (v1v) [inline catalog-info.yaml:685]: > @v1v the pipeline will be triggered remotely. Removed other branches from configuration just in case

If that's the case, what's the reason for using all those settings in the `provider_settings:` rather than:

```
      provider_settings:
        trigger_mode: none # don't trigger jobs from github activity
```

See https://github.com/elastic/beats/blob/051d06c17ac7a311649c022da31a5884d3e934f1/catalog-info.yaml#L90C1-L91C69

and https://registry.terraform.io/providers/buildkite/buildkite/latest/docs/resources/pipeline#trigger_mode

File: catalog-info.yaml [hunk 9]
```diff
@@ -648,13 +648,57 @@ spec:
...
+---
+# yaml-language-server: $schema=https://gist.githubusercontent.com/elasticmachine/988b80dae436cafea07d9a4a460a011d/raw/e57ee3bed7a6f73077a3f55a38e76e40ec87a7cf/rre.schema.json
+apiVersion: backstage.io/v1alpha1
+kind: Resource
+metadata:
+  name: buildkite-pipeline-elastic-agent-package-7-17
+  description: Buildkite pipeline for packaging Elastic Agent v7.17 package
+  links:
+    - title: Pipeline
+      url: https://buildkite.com/elastic/beats-xpack-elastic-agent-package-7-17
```
Comment (v1v) [inline catalog-info.yaml:671]: This name does not match the value in `spec.implementation.metadata.name`

File: catalog-info.yaml (hunk 8 above)
Comment (pazone) [reply catalog-info.yaml:685]: Good question. I've seen this example before and it is confusing. It uses the `trigger_mode: none` and all other settings as well. I'd use just `trigger_mode: none` without all other settings for github

Comment (pazone) [reply catalog-info.yaml:685]: Then probably we don't need `cancel_intermediate_builds` and `skip_intermediate_builds`

Comment (v1v) [reply catalog-info.yaml:685]: IIRC, you can configure the branch where the build can run to, see https://buildkite.com/docs/pipelines/trigger-step#environment-variables

File: catalog-info.yaml [hunk 10]
```diff
@@ -648,13 +648,48 @@ spec:
...
+      url: https://buildkite.com/elastic/beats-xpack-elastic-agent-package
+spec:
+  type: buildkite-pipeline
+  owner: group:ingest-fp
+  system: buildkite
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: beats-xpack-elastic-agent-package
```
Comment (v1v) [inline catalog-info.yaml:680]: does it make sense to include the branch name in the pipeline name? Otherwise, the name of the pipeline will not be shown unless accessing to the Buildkite UI that will show the description

File: .buildkite/x-pack/elastic-agent/pipeline.xpack.elastic-agent.package.yml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
```
Comment (v1v) [inline .buildkite/x-pack/elastic-agent/pipeline.xpack.elastic-agent.package.yml:1]: If only for `7.17`, I'd suggest this file should not be committed in this PR, but create a new PR targeting `7.17` with this file.

Otherwise, if someone uses the pipeline with the default branch `main` the build will not fail but will print hello?

**Review by v1v — APPROVED** [review]

**Comment by pazone** [comment]
@belimawr all set. could you take a look?

**Review by belimawr — APPROVED** [review]
